### PR TITLE
[IMP] mail: turn "Rtc/pingInterval" into field

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2734101

### CODE REVIEW

File: addons/mail/static/src/models/rtc/rtc.js [hunk 1]
```diff
@@ -1228,6 +1230,17 @@ registerModel({
...
         }),
+        /**
+         * Call all sessions for which no peerConnection is established at
+         * a regular interval to try to recover any connection that failed
+         * to start.
+         *
+         * This is distinct from this._recoverConnection which tries to restores
+         * connection that were established but failed or timed out.
+         */
+        intervalId: attr({
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:1241]: name more explicit to what it does

File: addons/mail/static/src/models/rtc/rtc.js [hunk 2]
```diff
@@ -1253,6 +1255,17 @@ registerModel({
...
         }),
+        /**
+         * Call all sessions for which no peerConnection is established at
+         * a regular interval to try to recover any connection that failed
+         * to start.
+         *
+         * This is distinct from this.recoverConnection which tries to restores
+         * connection that were established but failed or timed out.
+         */
+        missingRecoverInterval: attr({
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:1266]: I find the "missing" name to be misleading, and it's strange to talk about a recover for something that hasn't happened yet.

The idea is probably in the doc, maybe "establish", or even just "connect"?

File: addons/mail/static/src/models/rtc/rtc.js [hunk 3]
```diff
@@ -429,6 +411,13 @@ registerModel({
                 this._callPeer(session.peerToken);
             }
         },
+        /**
+         * @private
+         * @returns {Integer}
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:416]: lowercase? (and technically it's number)

File: addons/mail/static/src/models/rtc/rtc.js [hunk 4]
```diff
@@ -1208,6 +1210,17 @@ registerModel({
...
+         * Call all sessions for which no peerConnection is established at
+         * a regular interval to try to recover any connection that failed
+         * to start.
+         *
+         * This is distinct from this._recoverConnection which tries to restores
+         * connection that were established but failed or timed out.
+         */
+        connectInterval: attr({
+            compute: '_computeConnectInterval',
+        }),
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:1223]: I tested and it still works, but this comment is partially incomplete.

The timeout also does the server ping every 30s no matter what, which serves to ensure all new sessions are eventually known, even if they previously failed to fetch due to race conditions and other issues.

Comment (seb-odoo) [reply addons/mail/static/src/models/rtc/rtc.js:1223]: It's currently only documenting the "_callSessions" part but not the "_pingServer" part. I think both should be documented the same.

Comment (seb-odoo) [reply addons/mail/static/src/models/rtc/rtc.js:1223]: The ping is also useful to eject "timed out" members, and that's very not obvious from the code or the name itself.

Comment (alexkuhn) [reply addons/mail/static/src/models/rtc/rtc.js]: Just documented `ping`, and also renamed the interval to `pingInterval`, as the "ping" aspect is the most important one.
I really don't like the wording "call [peer/session]", so instead I use "connect to [RtcSession]".
This reduces the amount of concepts and makes it much easier to understand.

In follow-up PRs, most things will be in `RtcSession` and it's all actions on the `RtcSession`.
"Peer" will be replaced to "RtcSession" altogether. Just `RtcPeerConnection` will be technical detail of `RtcSession`, but we'll avoid using "Peer" in documentation and field naming.

**Review by phenix-factory — APPROVED** [review]

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**